src/fix_bd_sphere.cpp

```diff
@@ -0,0 +1,303 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+/* ----------------------------------------------------------------------
+   Originally modified from USER-CGDNA/fix_nve_dotc_langevin.cpp. 
+
+   Contributing author: Sam Cameron (University of Bristol)
+------------------------------------------------------------------------- */
+
+#include "fix_bd_sphere.h"
+
+#include <cmath>
+#include <cstring>
+#include "math_extra.h"
+#include "atom.h"
+#include "force.h"
+#include "update.h"
+#include "comm.h"
+#include "domain.h"
+#include "random_mars.h"
+#include "memory.h"
+#include "error.h"
+
+
+using namespace LAMMPS_NS;
+using namespace FixConst;
+
+/* ---------------------------------------------------------------------- */
+
+FixBdSphere::FixBdSphere(LAMMPS *lmp, int narg, char **arg) :
+  Fix(lmp, narg, arg)
+{
+  virial_flag = 1;
+
+  time_integrate = 1;
+  
+  if (narg != 8 && narg != 10)
+    error->all(FLERR,"Illegal fix bd/sphere command.");
+
+  if (!atom->sphere_flag)
+    error->all(FLERR,"Fix bd/sphere requires atom style sphere");
+  if (!atom->mu_flag)
+    error->all(FLERR,"Fix bd/sphere requires atom attribute mu");
```

Comment:
why this? is there nothing intrinsic in brownian dynamics for extended spherical objects that requires a dipole.
please see the source for fix nve/sphere how to keep support for dipoles with such an integrator optional.

Reply:
Yes good point. I had a specific system in mind here, I will make the changes.